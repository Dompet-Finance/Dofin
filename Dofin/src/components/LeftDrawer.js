'use strict'

import React from 'react'

import {
  DrawerLayoutAndroid,
  View,
  Text,
  Button,
} from 'react-native'

export class LeftDrawer extends React.Component {
  constructor(props) {
    super(props)
  }

  fakeNavigate() {
    return true
  }

  render() {
    var navigationView = (
      <View style={{flex: 1, backgroundColor: '#fff'}}>
        <Text style={{margin: 10, fontSize: 15, textAlign: 'left'}}>
          Im Drawer
        </Text>
      </View>
    )

    return (
      <DrawerLayoutAndroid
        drawerWidth={300}
        renderNavigationView={() => navigationView}>
        <View style={{flex: 1, alignItems: 'center'}}>
          <Text>
            Profile
          </Text>
          <Button
            onPress={() => this.fakeNavigate()}
            title="Learn More"
            color="#841584"
          />
          <Button
            onPress={() => this.fakeNavigate()}
            title="Learn More"
            color="#841584"
          />
          <Button
            onPress={() => this.fakeNavigate()}
            title="Learn More"
            color="#841584"
          />
        </View>
      </DrawerLayoutAndroid>
    )
  }
}

const styles = {
}

export default LeftDrawer