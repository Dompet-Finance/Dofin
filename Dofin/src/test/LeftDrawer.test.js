'use strict'

import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
// import { shallow } from 'enzyme'

import LeftDrawer from './LeftDrawer.js'


describe('Test LeftDrawer component', () => {

  test('renders correctly', () => {
    // const wrapper = shallow(
    //   <ListAvatarExample />
    // )
    const tree = renderer.create(
      <LeftDrawer />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  })

})

