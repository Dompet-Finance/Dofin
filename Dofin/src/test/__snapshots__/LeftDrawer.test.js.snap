// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test LeftDrawer component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "flex-start",
        "borderColor": "red",
        "borderWidth": 1,
      },
      undefined,
    ]
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Profile
    </Text>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Dashboard
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Transactions
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Dreams
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Setting
        </Text>
      </View>
    </View>
  </View>
</View>
`;
